source/docs/software/basic-programming/java-units.rst

```diff
@@ -68,13 +117,28 @@ Unit conversions can be done by calling ``Measure#in(Unit)``. The Java type syst
      maxAcceleration.in(MetersPerSecondPerSecond)
    );
 
-.. note:: Due to restrictions of the Java type system, acceleration is modeled as ``Velocity<Velocity<X>>>`` instead of having its own class.
```

Comment:
This note isn't sufficient either, but I think it's important to explain or at least mention `Velocity`.

Reply:
This is not my note, it was already there.
I agree, I had to read it three times before I understood `X` was actually `Velocity` in the example. If such a note is valuable, it probably belongs under "Using the Units Library". because it applies to all of the crazy looking types like `Measure<Per<Voltage, Velocity<Velocity<Distance>>>>`

Reply:
Yeah, that would probably be a good place to put it. Possibly with an explanation that it's shorthand for `Per<Time, _>`.

Reply:
`Velocity` isn't shorthand. It's incompatible with `Per` because of limitations in the Java type system. The `Unit` class takes the self type as a generic argument, meaning `Velocity<D extends Unit<D>> extends Per<D, Time>` would be out of bounds.

Kotlin teams can use type aliasing and may be able to completely ignore the `Velocity` class, but that's out of scope for our documentation